### DISCUSSION TITLE
Add the option to set css class to Nuxt root element at nuxt.config

### DISCUSSION THREAD

**Discussion by victorlmneves** [nuxt/nuxt#26609] [Ideas]
Hi all
First of all, thank you for all your hard work in providing such a cool framework

I'm currently working on a migration of a Vue 3 SSR project to Nuxt 3
When we started the project the idea was to use Nuxt but at that time (2 years ago), Nuxt 3 wasn't yet released.
But having that in mind, our folder structure was already based on what we knew about Nuxt to mitigate the effort for the migration.

But we forgot the small detail that Nuxt doesn't have the index.html and when working on the styles we used the root element to set the grid.

Vue index.html
```
<body ssr-bodyAttrs class="base-body">
    <div id="app" class="base-wrapper"><!--ssr-app-html--></div>
    <!--ssr-initial-state-->
    <script type="module" src="/src/entry-client.ts"></script>
</body>
```

This has forced us to rework our CSS, renaming that class to an ID and using the rootId option on Nuxt config to inject as the ID.
Based on this, my suggestion is to extend the Nuxt config by providing something like a `rootClass` or `rootCssClass` as this would prevent anyone migrating to Nuxt from any technology where it has a CSS class or classes applied there, being forced to refactor the CSS, that is some cases can be a low effort but it can also be a high effort depending on multi-factors

nuxt.config
```
app: {
    keepalive: true,
    head: {
        ...
    },
    rootId: 'app',
    rootClass: 'base-wrapper',
},
```

Thanks

**Comment by dsvgl** [comment]
This is possible with https://nuxt.com/docs/api/composables/use-head

example:
```ts
useMeta({
  bodyAttrs: {
    class: 'foobar',
  },
});
```

I think the same is possible to directly set within your nuxt.config.ts
```ts
app: {
  head: {
    bodyAttrs: {
      class: 'foobar',
    },
  },
},
```

**Reply by victorlmneves** [reply]
@dsvgl 
I'm not talking about adding to the body tag
It's for the entry point element from Nuxt
```
<body>
  <div id="__nuxt">content will injected here</div> // to add the CSS class here
</body>
```

**Reply by dsvgl** [reply]
ah ok. Maybe this could rather be an feature request issue :)

**Comment by weo3dev** [comment]
@victorlmneves  - I think this may be what you are looking for? https://nuxt.com/docs/api/nuxt-config#roottag  -- to be fair, this was certainly not included in RC and launch version of Nuxt 3, but it does look like it is now available, along with being able to "hide" the default '__nuxt' identifying class (which other colleagues of mine needed for privacy purposes on their builds). Hope this helps.

**Reply by victorlmneves** [reply]
Don't think so
My understanding of that setting is to define a new element as root, but I don't think I can set a CSS class on the element

**Reply by RomainMazB** [reply]
@weo3dev was close.

Here is what you are searching for: https://nuxt.com/docs/api/nuxt-config#rootattrs

**Comment by masyoudi** [comment]
same question [here](https://github.com/nuxt/nuxt/discussions/24447#discussioncomment-7662285)

**Comment by MuhammadM1998** [comment]
This is added in https://github.com/nuxt/nuxt/pull/27014. Wait for the next release or use the edge version to use it
